src/lib/reasoners/satml.ml

```diff
@@ -1027,10 +1039,27 @@ module Make (Th : Theory.S) : SAT_ML with type th = Th.t = struct
            assert (ta.is_true);
            assert (ta.var.level >= 0);
            if ta.var.level = 0 then begin
-             incr nb_f;
              (ta.lit, Th_util.Other, Ex.empty, 0, env.cpt_current_propagations)
              :: acc
            end
+           (* This was first added by PR 1000, but it triggers a bug in
+              tests/cc/testfile-ac_cc002.ae with option
+              --no-tableaux-cdcl-in-instantiation *)
```

Comment:
I think it was important to restore the old unit_tenv from before a push (cf Satml.pop)

Reply:
I believe this should only be the case with the code below uncommented — without this code, Alt-Ergo should only ever add to `unit_tenv` facts that are true at level `0` (otherwise it is probably a soundness bug) which shouldn't be impacted by push/pop.

(If I recall correctly, I added this code and the `unit_tenv` stack because `unit_tenv` with the specification of only adding things known at level `0` is less useful with incremental solving and we were trying to solve the performance issues)